tests/next_tests/integration_tests/feature_tests/cases.py

```diff
@@ -0,0 +1,358 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+import dataclasses
+import functools
+import inspect
+import types
+import typing
+from typing import Any, Callable, Optional, Protocol, TypeAlias
+
+import numpy as np
+import pytest
+from typing_extensions import Self
+
+from gt4py.next import common
+from gt4py.next.ffront import decorator, fbuiltins
+from gt4py.next.iterator import embedded, ir as itir
+from gt4py.next.program_processors import processor_interface as ppi
+from gt4py.next.type_system import type_specifications as ts, type_translation
+
+from next_tests.integration_tests.feature_tests.ffront_tests.ffront_test_utils import (  # noqa: F401 # import alias for later moving this here; noqa: F401: # fixture
+    Edge,
+    IDim,
+    Ioff,
+    JDim,
+    Joff,
+    KDim,
+    Koff,
+    Vertex,
+    fieldview_backend,
+)
+
+
+IField: TypeAlias = common.Field[[IDim], int]
+IJKField: TypeAlias = common.Field[[IDim, JDim, KDim], int]
+IJKFloatField: TypeAlias = common.Field[[IDim, JDim, KDim], np.float64]
+VField: TypeAlias = common.Field[[Vertex], int]
+EField: TypeAlias = common.Field[[Edge], int]
+V2EDim = common.Dimension("V2E", kind=common.DimensionKind.LOCAL)
```

Comment:
We discussed this and concluded to avoid changing anything in `reduction_setup` for now. @DropD Please add the todo here that the duplication of `V2EDim`, `E2VDim`, `V2E`, `E2V` should be removed.